www/apps/book/app/learn/fundamentals/module-links/index-module/page.mdx

```diff
@@ -0,0 +1,487 @@
+export const metadata = {
+  title: `${pageNumber} Index Module`,
+}
+
+# {metadata.title}
+
+In this chapter, you'll learn about the Index Module and how you can use it.
+
+<Note type="soon" title="Experimental">
+
+The Index Module is an experimental feature that's subject to change. It is not recommended for production use.
+
+</Note>
+
+## What is the Index Module?
+
+The Index Module adds a method to [Query](../query/page.mdx) that allows you to filter data by linked modules.
+
+While modules share the same database by default, Medusa [isolates modules](../../modules/isolation/page.mdx) to allow using external data sources or different database types.
+
+So, when you retrieve data across modules using Query, Medusa aggregates the data coming from diffeent modules to create the end result. This approach limits your ability to filter data by linked modules. For example, you can't filter products (created in the Product Module) by their brand (created in the Brand Module).
+
+The Index Module solves this problem by injesting data into a central data store on application start up. The data store has a relational structure that enables efficiently filtering the data injested from different modules (and their data stores). So, when you retrieve data with the Index Module, you're retrieving them from the injested data store, not the original data source.
+
+![Diagram showcasing how data is retrieved from the Index Module's data store](https://res.cloudinary.com/dza7lstvk/image/upload/v1747988533/Medusa%20Book/index-module_epurmt.jpg)
+
+### Injested Data Models
+
+Currently, only the following data models are injested by the Index Module when you install it:
+
+- `Product`
+- `ProductVariant`
+- `Price`
+- `SalesChannel`
+
+Future versions may add more data models to the list.
+
```

Comment:
todo: Should we explain what this means?

For example:

> Consequently, you can only index custom data models if they are linked to one of these four. We are continuously adding more core entities to the Index, as we see the demand for it.

Reply:
I realise this is mentioned later, but I still think we can add a short comment here and link to the section below.